moveToWebcompat.js

```diff
@@ -86,42 +111,89 @@ const sendToWebcompat = async bug => {
     steps_reproduce: steps
   };
 
-  const form = convertToFormData(data);
-
   return await fetch(WEBCOMPAT_ENDPOINT, {
     method: "POST",
-    body: form
+    body: convertToFormData(data)
   });
 };
 
 const onMoveButtonClick = async () => {
   const bugId = getBugId();
-  if (!bugId)
-    openDropDown(
+  if (!bugId) {
+    return showError(
       "A valid bug ID could not be found. Please check you're actually viewing a bug."
     );
+  }
 
-  const result = await getBugData(bugId);
+  startLoadingState();
+
+  const result = await getBugData(bugId).catch(err => new Error(err));
+
+  if (result instanceof Error) {
+    return showError(
+      "There is an error with retrieving bug data from bugzilla."
+    );
+  }
 
   if (result.bugs && result.bugs.length) {
-    const required = getRequired(result.bugs[0]);
+    const bug = result.bugs[0];
+
+    if (bug.status === "RESOLVED") {
+      return showError(
+        "The bug has already been resolved. Please reopen it and try again."
+      );
+    }
+
+    const required = getRequired(bug);
     if (required.length) {
-      openDropDown("Please fill in the following:", required);
-      return;
+      stopLoadingState();
+      return openDropDown("Please fill in the following:", required);
+    }
+
+    const response = await sendToWebcompat(bug).catch(err => new Error(err));
+
+    if (response instanceof Error) {
+      return showError(
+        "There is an error with importing the bug to webcompat.com. Please try to create it manually."
+      );
     }
 
-    const response = await sendToWebcompat(result.bugs[0]);
+    if (response.url) {
+      stopLoadingState();
+      resolveAsMoved(response.url);
+    }
   }
 };
 
-const buildButton = () => {
-  const moveButton = document.createElement("button");
-  moveButton.classList.add("secondary");
-  moveButton.type = "button";
-  moveButton.title = "Import bug to webcompat.com and resolve issue as Moved";
-  moveButton.innerText = "Move to webcompat.com";
+const fillSeeAlso = issueLink => {
+  document.getElementById("mode-btn").click();
+  document.getElementById("see_also-btn").click();
+  document.getElementById("see_also").value = issueLink;
+};
+
+const resolveAsMoved = issueLink => {
+  fillSeeAlso(issueLink);
+
+  [...document.getElementById("resolve-as").children]
+    .filter(el => el.textContent.includes("FIXED"))[0]
+    .click();
+
+  const resolution = document.getElementById("bottom-resolution");
+  resolution.value = "MOVED";
+  resolution.dispatchEvent(new Event("change"));
 
-  return moveButton;
+  setTimeout(() => document.getElementById("bottom-save-btn").click(), 100);
```

Comment:
Why must we use a `setTimeout` here? If it's to wait for the change event above to be handled first, it seems racy.

Reply:
Mmm it's not really needed here, didn't remove the timeout after testing. 

Initially I have been trying to change the select value by just setting `resolution.value = "MOVED";`, but it didn't get saved. I wasn't sure why, so tried to add a timeout, but it wasn't the reason. Tried dispatching change event and it worked as bugzilla's js script is expecting a change event to save the value (the keep track of "dirty" fields, and if the field hasn't been touched, they override it to be the initial value).